### PR TITLE
Modify rule S1913: LaYC format

### PR DESCRIPTION
**Pull request by marco-antognini-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [x] logical errors and incorrect information
- [x] information gaps and missing content
- [x] text style and tone
- [x] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S1913/cfamily/rule.adoc [hunk 1]
```diff
@@ -1,11 +1,20 @@
 == Why is this an issue?
 
-A call to ``++sizeof(sizeof(...))++`` is equivalent to ``++sizeof(size_t)++``, and indicates a misuse or misunderstanding of the ``++sizeof++`` construct.
+Given an expression `e` of type `T`, `sizeof(e)` returns the size in bytes of `T`, without evaluating `e`.
+Furthermore, the expression `sizeof(e)` has type `size_t`.
 
+This means that `sizeof(sizeof(e))` is equivalent to `sizeof(size_t)`.
+In other words, its value does not depend on `e` nor has any side effects.
```
Comment (amelie-renard-sonarsource) [inline rules/S1913/cfamily/rule.adoc:7]: "nor has any side effects": I find this information necessary and think we should emphasize it more.
The way I see it is that the user likely made a mistake because:
- sizeof(sizeof()) always returns the same result and does not depend on `e`
- sizeof does not evaluate the expression, which means that there is no side effect

I would separate these two things to make the second one more explicit than it currently is. WDYT?

**Review by amelie-renard-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S1913/cfamily/rule.adoc (hunk 1 above)
Comment (marco-antognini-sonarsource) [reply rules/S1913/cfamily/rule.adoc]: Great idea. I rephased the description in bdd67d6213. Let me know what you think.

File: rules/S1913/cfamily/rule.adoc [hunk 2]
```diff
@@ -1,45 +1,52 @@
 == Why is this an issue?
 
-A call to ``++sizeof(sizeof(...))++`` is equivalent to ``++sizeof(size_t)++``, and indicates a misuse or misunderstanding of the ``++sizeof++`` construct.
+Given an expression `e` of type `T`, `sizeof(e)` returns the size in bytes of `T`, without evaluating `e`.
+Furthermore, the expression `sizeof(e)` has type `size_t`.
+This means that `sizeof(sizeof(e))` is equivalent to `sizeof(size_t)`.
```
Comment (amelie-renard-sonarsource) [inline rules/S1913/cfamily/rule.adoc:5]: What about:
```suggestion
Given an expression `e` of type `T`, `sizeof(e)` returns the size in bytes of `T`. The `sizeof` function does not evaluate the expression `e`, and the result of `sizeof(e)` is of type `size_t`. Therefore, `sizeof(sizeof(e))` is equivalent to `sizeof(size_t)`.
```

File: rules/S1913/cfamily/rule.adoc [hunk 3]
```diff
@@ -1,45 +1,52 @@
...
 
+In other words:
```
Comment (amelie-renard-sonarsource) [inline rules/S1913/cfamily/rule.adoc:7]: And then, I would keep:
```suggestion
In other words, `sizeof(sizeof(e))` always returns the same result and does not depend on `e`, which is unlikely what was expected.
```

File: rules/S1913/cfamily/rule.adoc (hunk 1 above)
Comment (amelie-renard-sonarsource) [reply rules/S1913/cfamily/rule.adoc]: I find the result better but still a bit hard to follow. I wrote down suggestions, let me know what you think.

File: rules/S1913/cfamily/rule.adoc (hunk 2 above)
Comment (marco-antognini-sonarsource) [reply rules/S1913/cfamily/rule.adoc]: Thanks for the suggestion.

Pedantic nit: `sizeof` is an operator, not a function. Although the difference does not really matter here, I've adapted your suggestion.

I also changed it slightly in 8771cba278 to have one statement/fact per sentence. I've also put more emphasis on the "no side effect" parts.

File: rules/S1913/cfamily/rule.adoc [hunk 4]
```diff
@@ -1,45 +1,50 @@
 == Why is this an issue?
 
-A call to ``++sizeof(sizeof(...))++`` is equivalent to ``++sizeof(size_t)++``, and indicates a misuse or misunderstanding of the ``++sizeof++`` construct.
+Given an expression `e` of type `T`, `sizeof(e)` returns the size in bytes of `T`.
+The `sizeof` operator results in a value of `size_t`.
```
Comment (amelie-renard-sonarsource) [inline rules/S1913/cfamily/rule.adoc:4]: I am not used to this wording, so it seems strange to me.
What about:
```suggestion
The `sizeof` operator results in a value of type `size_t`.
```
Or
```suggestion
The return type of the `sizeof` operator is `size_t`.
```

**Review by amelie-renard-sonarsource — APPROVED** [review]
Small suggestion, otherwise LGTM
